upi/vsphere/network/cidr_to_ip.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# cidr_to_ip - 
+#  https://www.terraform.io/docs/providers/external/data_source.html
+#  Based on info from here: https://gist.github.com/irvingpop/968464132ded25a206ced835d50afa6b
+#  This script takes the CIDR address and cycles through looking for the first available address
+#  echo '{"cidr": "139.178.89.192/26", "master_count": "3", "worker_count": "3", "cluster_domain": "dphillip.devcluster.openshift.com"}' | ./cidr_to_ip.sh
+function error_exit() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+function check_deps() {
+  test -f $(which jq) || error_exit "jq command not detected in path, please install it"
+  test -f $(which ipcalc) || error_exit "ipcalc command not detected in path, please install it"
+  test -f $(which dig) || error_exit "dig command not detected in path, please install it"
+
+}
+
+function parse_input() {
+  # jq reads from stdin so we don't have to set up any inputs, but let's validate the outputs
+  eval "$(jq -r '@sh "export CIDR=\(.cidr) master_count=\(.master_count)  worker_count=\(.worker_count) cluster_domain=\(.cluster_domain)"')"
+  if [[ -z "${CIDR}" ]]; then export CIDR=none; fi
+  if [[ -z "${master_count}" ]]; then export master_count=none; fi
+  if [[ -z "${worker_count}" ]]; then export worker_count=none; fi
+  if [[ -z "${cluster_domain}" ]]; then export cluster_domain=none; fi
+}
+
+function produce_output() {
+
+  cidr=$CIDR
+
+  # range is bounded by network (-n) & broadcast (-b) addresses.
+  lo=$(ipcalc -n $cidr | cut -f2 -d=)
+  hi=$(ipcalc -b $cidr | cut -f2 -d=)
+
+  read a b c d <<< $(echo $lo | tr . ' ')
+  read e f g h <<< $(echo $hi | tr . ' ')
+  IP_RANGE=$(eval echo {$a..$e}.{$b..$f}.{$c..$g}.{$d..$h})
+   
+  bs_count=0
+  m_count=0
+  w_count=0
+
+  # check cluster_domain DNS first
+  for ETCD in 0 1 2
+  do
+	
+        DNS_RECORDS+="$(dig +short etcd-$ETCD.${cluster_domain}) " 
```

Comment:
Do you mind rebuilding on top of https://github.com/openshift/installer/pull/1492? Then we can search for control-plane-{0,1,2}.${cluster_domain} and compute-{0,1,2}.${cluster_domain} instead of etcd.